### ISSUE TITLE
Docker CPI - Can't find property 'docker_cpi.docker.tls.certificate'

### ISSUE THREAD

**Issue by ivandavidov** [cloudfoundry/bosh-deployment#93]
Hello,

I've been trying to setup Bosh director on Docker and due to lack of installation instructions I used the command provided [here](https://github.com/cloudfoundry/bosh-deployment/blob/master/test.sh#L421):

```
bosh create-env bosh.yml \
  -o docker/cpi.yml \
  -o jumpbox-user.yml \
  --state=$vars_store_prefix \
  --vars-store $(mktemp ${vars_store_prefix}.XXXXXX) \
  -v director_name=docker \
  -v internal_cidr=10.245.0.0/16 \
  -v internal_gw=10.245.0.1 \
  -v internal_ip=10.245.0.10 \
  -v docker_host=tcp://192.168.50.8:4243 \
  -v docker_tls=ca_cert \
  -v network=net3
```

I issued the command exactly as it is provided. The only thing I changes was the call to ``bosh`` because on my machine I installed it as ``bosh2`` in order to be able to work both with v1 and v2 CLI.

This is the output I get each time I execute the command.

```
+ bosh2 create-env bosh.yml -o docker/cpi.yml -o jumpbox-user.yml --state= --vars-store .wZPieL -v director_name=docker -v internal_cidr=10.245.0.0/16 -v internal_gw=10.245.0.1 -v internal_ip=10.245.0.10 -v docker_host=tcp://192.168.50.8:4243 -v docker_tls=ca_cert -v network=net3
Deployment manifest: '/FS/fslocal/cloudfoundry/bosh-deployment/bosh.yml'
Deployment state: '/FS/fslocal/cloudfoundry/bosh-deployment/bosh-state.json'

Started validating
  Downloading release 'bosh'... Skipped [Found in local cache] (00:00:00)
  Validating release 'bosh'... Finished (00:00:02)
  Downloading release 'bosh-docker-cpi'... Skipped [Found in local cache] (00:00:00)
  Validating release 'bosh-docker-cpi'... Finished (00:00:02)
  Downloading release 'os-conf'... Skipped [Found in local cache] (00:00:00)
  Validating release 'os-conf'... Finished (00:00:00)
  Validating cpi release... Finished (00:00:00)
  Validating deployment manifest... Finished (00:00:00)
  Downloading stemcell... Skipped [Found in local cache] (00:00:00)
  Validating stemcell... Finished (00:00:02)
Finished validating (00:00:08)

Started installing CPI
  Compiling package 'golang_1.7/21609f611781e8586e713cfd7ceb389cee429c5a'... Finished (00:00:00)
  Compiling package 'docker_cpi/ab58317d13c89566a161b40115de222b80474739'... Finished (00:00:00)
  Installing packages... Finished (00:00:02)
  Rendering job templates... Failed (00:00:00)
Failed installing CPI (00:00:02)

Installing CPI:
  Rendering and uploading Jobs:
    Rendering job templates for installation:
      Rendering templates for job 'docker_cpi/ea5bcf51e5da2233b03c97ec118caa0ec7b716eb':
        Rendering template src: cpi.json.erb, dst: config/cpi.json:
          Rendering template src: /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/bosh-release-job582547316/templates/cpi.json.erb, dst: /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/rendered-jobs826377360/config/cpi.json:
            Running ruby to render templates:
              Running command: 'ruby /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/erb-renderer227669177/erb-render.rb /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/erb-renderer227669177/erb-context.json /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/bosh-release-job582547316/templates/cpi.json.erb /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/rendered-jobs826377360/config/cpi.json', stdout: '', stderr: '/home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/erb-renderer227669177/erb-render.rb:189:in `rescue in render': Error filling in template '/home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/bosh-release-job582547316/templates/cpi.json.erb' for docker_cpi/0 (line 26: #<TemplateEvaluationContext::UnknownProperty: Can't find property 'docker_cpi.docker.tls.certificate'>) (RuntimeError)
        from /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/erb-renderer227669177/erb-render.rb:175:in `render'
        from /home/apama/.bosh/installations/7e5e7c36-e77e-4599-7a17-db143a0860d6/tmp/erb-renderer227669177/erb-render.rb:200:in `<main>'
':
                exit status 1

Exit code 1
```

What bothers me is the message ``Can't find property 'docker_cpi.docker.tls.certificate'``. The only reference to this property that I could find is in [this pull request](https://github.com/cppforlife/bosh-docker-cpi-release/pull/2/files) merged by Dmitriy Kalinin.

Could you please assist me and elaborate on the necessary prerequisites I need to fulfill before I run the ``create-env`` command? I have the feeling that I might need to generate certificate keypair and provide the path(???) to the keypair in [this line](https://github.com/cloudfoundry/bosh-deployment/blob/master/test.sh#L431). Currently this is my best guess.

Long story short - I'm trying to run Bosh director via Docker CPI and I got the problem described above. All assistance on this matter is highly appreciated. Thank you!

Regards,
Ivan Davidov

**Comment by ivandavidov** [comment]
@jtarchie  - I'm using the latest available Git sources and in ``docker/cpi.yml`` I see the following:

```
- type: replace
  path: /releases/-
  value:
    name: bosh-docker-cpi
    version: 0.0.3
    url: https://bosh.io/d/github.com/cppforlife/bosh-docker-cpi-release?v=0.0.3
    sha1: 9eb1457a5f268a8e3a21765d151366e28f7d5ea3
```

I guess this means I'm using version ``0.0.3``.

**Comment by ivandavidov** [comment]
The issue might be due to me being quite new to both ``bosh`` and ``docker``. I just learned that the docker daemon can be secured and most probably I need to provide the certificate for the secure communication with the docker daemon. However, my daemon is currently not configured in such way which means that first of all I have to secure my docker daemon properly.

The real issue is that there are no instructions at all how to properly set bosh + docker and it takes me quite a lot of time to reverse engineer what should happen. It would be great if you guys provide at least high level instructions on this topic somewhere in the bosh docs.

**Comment by jfmyers9** [comment]
Closing this due to inactivity.

If you would like to discuss this contribution further, please open a new issue.
